neo/Consensus/ConsensusContext.cs

```diff
@@ -73,10 +73,10 @@ public void Deserialize(BinaryReader reader)
             PrevHash = reader.ReadSerializable<UInt256>();
             BlockIndex = reader.ReadUInt32();
             ViewNumber = reader.ReadByte();
-            Validators = reader.ReadSerializableArray<ECPoint>(byte.MaxValue);
+            Validators = reader.ReadSerializableArray<ECPoint>(ConsensusService.MaxValidatorsCount);
```

Comment:
In fact, after running `Reset()`, we have accurately known the current number of validators.

Reply:
It can be smarter; since there are cases in the future where this may be deserialized without having reset, it could use the Validators if they are non-null.

Reply:
Right now it is always deserialized before Reset though. I think it is fine to leave it as is here.

Reply:
The data you deserialize is read from the `store`, and the data is written after you serialize it, and the serialized data is derived from `Reset()`. So deserialization does not require a security check.

Reply:
In this way, all other fields that match the number of validators need to be compared to `Validators.Length`.

Reply:
So we can remove `MaxValidatorsCount`.

Reply:
If you don't mind, I will modify this place directly.

Reply:
Please do.

Reply:
However, please make sure RecoveryMessage also uses the right max.

Reply:
Maybe we can set it in a static that `RecoveryMessage` can use when deserializing or something.

Reply:
@erikzhang Any ETA on your changes for modifying this?

Reply:
It's too late here. I will do it tomorrow.

Reply:
Ok. It is 4AM here. I will go to bed ;-)